src/main/java/webserver/http/request/HttpRequestParser.java

```diff
@@ -4,59 +4,169 @@
-            throw new IllegalArgumentException("Empty request line");
+    private static String readHeaderString(InputStream in) throws IOException {
+        byte[] buffer = new byte[1];
+        StringBuilder headerBuilder = new StringBuilder();
+
+        boolean foundHeaderEnd = false;
+        while (!foundHeaderEnd) {
+            int bytesRead = in.read(buffer);
+            if (bytesRead == -1) {
+
+                throw new IOException("스트림의 끝에 도달했습니다.");
+            }
+
+            headerBuilder.append(new String(buffer, 0, bytesRead, StandardCharsets.UTF_8));
+
+            if (headerBuilder.length() >= 4 &&
+                    headerBuilder.substring(headerBuilder.length() - 4).equals(DOUBLE_CRLF)) {
+                foundHeaderEnd = true;
+            }
+        }
+
+        return headerBuilder.toString();
+    }
+
+    private static PostRequestResult processPostRequest(InputStream in,
+                                                        Map<String, List<String>> headers) throws IOException {
+        String body = null;
+        Map<String, List<String>> bodyParameters = new HashMap<>();
+
+        if (headers.containsKey(CONTENT_LENGTH)) {
+            int contentLength = Integer.parseInt(headers.get(CONTENT_LENGTH).get(0));
+
+            if (contentLength > 0) { // body가 있는 경우
+                body = readBody(in, contentLength);
+                logger.debug(body);
+                if (isFormUrlEncoded(headers)) {
+                    bodyParameters = parseQueryString(body);
+                }
+            }
         }
 
-        String[] requestLineArr = requestLine.split(" ");
+        return new PostRequestResult(body, bodyParameters);
+    }
+
+    private static Map<String, List<String>> parseQueryString(String queryString) {
+        Map<String, List<String>> parameters = new HashMap<>();
+        String[] pairs = queryString.split(AND);
+
+        for (String pair : pairs) {
+            int equalPos = pair.indexOf(EQUAL);
+            if (equalPos < 0) {
+                throw new IllegalArgumentException("유효하지 않은 파라미터 형식: " + pair);
+            }
+
+            String key = pair.substring(0, equalPos);
+            String value = pair.substring(equalPos+1);
+
+            key = FileUtils.decodeUrl(key);
+            value = FileUtils.decodeUrl(value);
+
+            parameters.computeIfAbsent(key, k -> new ArrayList<>()).add(value);
+        }
+        return parameters;
+    }
+
+    private static String readBody(InputStream in, int contentLength) throws IOException {
+        byte[] buffer = new byte[contentLength];
+        int bytesRead = 0;
+        int totalBytesRead = 0;
+
+        while (totalBytesRead < contentLength) {
+            bytesRead = in.read(buffer, totalBytesRead, contentLength - totalBytesRead);
+            if(bytesRead == -1) break;
+            totalBytesRead += bytesRead;
+        }
+
+        return new String(buffer, 0, totalBytesRead, StandardCharsets.UTF_8);
```

Comment:
```suggestion
        return new String(buffer, StandardCharsets.UTF_8);
```

요렇게 해도 될껄요? 그 외 상황은 예외처리해주면 될 것 같습니다.